cluster-bootstrap/src/main/scala/akka/management/cluster/bootstrap/ClusterBootstrap.scala

```diff
@@ -91,18 +93,13 @@ final class ClusterBootstrap(implicit system: ExtendedActorSystem) extends Exten
    * This allows us to "reverse lookup" from a lowest-address sorted contact point list,
    * that we discover via discovery, if a given contact point corresponds to our remoting address,
    * and if so, we may opt to join ourselves using the address.
-   *
-   * @return true if successfully set, false otherwise (i.e. was set already)
    */
   @InternalApi
   private[akka] def setSelfContactPoint(baseUri: Uri): Unit =
     _selfContactPointUri.success(baseUri)
 
   /** INTERNAL API */
-  @InternalApi private[akka] def selfContactPoint: Future[(String, Int)] =
-    _selfContactPointUri.future.map { uri =>
-      (uri.authority.host.toString, uri.authority.port)
-    }
+  @InternalApi private[akka] def selfContactPoint: Future[Uri] = _selfContactPointUri.future
```

Comment:
We already have a future here that resolves to the self contact-point. The mapping function is better refactored out where the future is currently used. Then we can re-use the future for our own purposes.